lxd/main_forkmknod.go

```diff
@@ -197,7 +199,12 @@ void forkmknod()
 		_exit(EXIT_FAILURE);
 	}
 
-	if (!same_fsinfo(&s1, &s2, &sfs1, &sfs2) || !(sfs2.f_flags & MS_NODEV)) {
+	if (!same_fsinfo(&s1, &s2, &sfs1, &sfs2) && !(sfs2.f_flags & MS_NODEV)) {
```

Comment:
Hmm, what's going on with this part of the change?

Doesn't that mean we'd go through if chk_perm_only=false and fsinfo returns true even though we're on NODEV?

Reply:
Fixed.